src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/support/DeadlineSupportController.java

```diff
@@ -0,0 +1,175 @@
+import java.time.Month;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+@RestController
+@RequestMapping("/support/deadline")
+public class DeadlineSupportController {
+    private static final String CLAIM_DOES_NOT_EXIST = "Claim %s does not exist";
+
+    private static final Predicate<Claim> CREATED_BEFORE_5_0_0 = claim -> claim.getCreatedAt()
+        .isBefore(LocalDateTime.of(2019, Month.SEPTEMBER, 9, 15, 12));
+    private static final Predicate<Claim> DEFENDANT_MEDIATION = claim -> claim.getResponse()
+        .flatMap(Response::getFreeMediation)
+        .filter(YesNoOption.YES::equals)
+        .isPresent();
+    private static final Predicate<Claim> FULL_ADMIT_RESPONSE = claim -> claim.getResponse()
+        .map(Response::getResponseType)
+        .filter(ResponseType.FULL_ADMISSION::equals)
+        .isPresent();
+    private static final Predicate<Claim> CLAIMANT_ACCEPTED = claim -> claim.getClaimantResponse()
+        .map(ClaimantResponse::getType)
+        .filter(ClaimantResponseType.ACCEPTATION::equals)
+        .isPresent();
+    private static final Predicate<Claim> CLAIMANT_MEDIATION = claim -> claim.getClaimantResponse()
+        .filter(claimantResponse -> ClaimantResponseType.REJECTION.equals(claimantResponse.getType()))
+        .map(ResponseRejection.class::cast)
+        .flatMap(ResponseRejection::getFreeMediation)
+        .filter(YesNoOption.YES::equals)
+        .isPresent();
+
+    private final UserService userService;
+    private final ClaimService claimService;
+    private final DirectionsQuestionnaireService dqService;
+
+    public DeadlineSupportController(
+        UserService userService,
+        ClaimService claimService,
+        DirectionsQuestionnaireService dqService
+    ) {
+        requireNonNull(userService);
+        requireNonNull(claimService);
+        requireNonNull(dqService);
+
+        this.userService = userService;
+        this.claimService = claimService;
+        this.dqService = dqService;
+    }
+
+    @PutMapping("/{deadlineType}/claim/{referenceNumber}")
+    @ApiOperation("Calculate and define a deadline for a claim")
+    @ApiResponses({
+        @ApiResponse(code = 200, message = "Deadline already defined"),
+        @ApiResponse(code = 201, message = "Deadline calculated and defined"),
+        @ApiResponse(code = 403, message = "Claim is in an invalid state for this deadline")
+    })
+    public ResponseEntity<String> defineDeadline(
+        @PathVariable String deadlineType,
+        @PathVariable String referenceNumber
+    ) {
+        String authorisation = userService.authenticateAnonymousCaseWorker().getAuthorisation();
+        Claim claim = claimService.getClaimByReferenceAnonymous(referenceNumber)
+            .orElseThrow(() -> new NotFoundException(format(CLAIM_DOES_NOT_EXIST, referenceNumber)));
+
+        if ("dq".equals(deadlineType)) {
```

Comment:
Deadline type should be an enum.

Reply:
When/If we ever add new deadline types we'll think about the best way to distinguish them. As it stands, the use of inline string constants is consistent with the claim-store generally (took me a while to get used to it...)